README.md

```diff
@@ -28,6 +34,9 @@ The Matchbox project contains both:
 
 ## How it works
 
+![Connection](images/connection.light.excalidraw.svg#gh-light-mode-only)
+![Connection](images/connection.dark.excalidraw.svg#gh-dark-mode-only)
```

Comment:
I like this diagram,

does rustdoc understand light/dark as well?

Reply:
If not, perhaps it could work to just use a somewhat mid to slighlty darker gray that is still visible on dark themes? (that's what the logo does)

Reply:
Rustdoc doesn't seem to understand images at all, my local build isn't picking up either this or the logo, but it's possible I'm doing something wrong

Reply:
It seems like there is no good way to do it at the moment. Either the images have to be hosted externally, or we could use a hack like https://docs.rs/embed-doc-image/latest/embed_doc_image/

There is a pretty rfc for it: https://github.com/rust-lang/rfcs/pull/3397

Reply:
We could use github pages to host our images... Would take a bit of hacking around with CI

Reply:
Realised we can actually just point to `raw.githubusercontent`. Only slight issue is that we won't be able to check it works until we merge :grimacing: (though you can change it to the old URL to assure yourself)

Reply:
imo better to keep images out of rustdoc until fully supported. Would be a better image for the README.

Reply:
They're the same, currently

Reply:
I think this is fine for now. It's also not the end of the world if it doesn't work or they don't appear. I'll make sure to check the readme after merging.